allennlp_models/vision/dataset_readers/vgqa.py

```diff
@@ -0,0 +1,258 @@
+    """
+
+    def __init__(
+        self,
+        image_dir: Optional[Union[str, PathLike]] = None,
+        *,
+        image_loader: Optional[ImageLoader] = None,
+        image_featurizer: Optional[Lazy[GridEmbedder]] = None,
+        region_detector: Optional[Lazy[RegionDetector]] = None,
+        answer_vocab: Optional[Union[Vocabulary, str]] = None,
+        feature_cache_dir: Optional[Union[str, PathLike]] = None,
+        tokenizer: Optional[Tokenizer] = None,
+        token_indexers: Optional[Dict[str, TokenIndexer]] = None,
+        cuda_device: Optional[Union[int, torch.device]] = None,
+        max_instances: Optional[int] = None,
+        image_processing_batch_size: int = 8,
+        write_to_cache: bool = True,
+    ) -> None:
+        run_featurization = image_loader and image_featurizer and region_detector
+        if image_dir is None and run_featurization:
+            raise ValueError(
+                "Because of the size of the image datasets, we don't download them automatically. "
+                "Please go to https://visualgenome.org/api/v0/api_home.html, download the datasets you need, "
+                "and set the image_dir parameter to point to your download location. This dataset "
+                "reader does not care about the exact directory structure. It finds the images "
+                "wherever they are."
+            )
+
+        super().__init__(
+            image_dir,
+            image_loader=image_loader,
+            image_featurizer=image_featurizer,
+            region_detector=region_detector,
+            feature_cache_dir=feature_cache_dir,
+            tokenizer=tokenizer,
+            token_indexers=token_indexers,
+            cuda_device=cuda_device,
+            max_instances=max_instances,
+            image_processing_batch_size=image_processing_batch_size,
+            write_to_cache=write_to_cache,
+        )
+
+        # read answer vocab
+        if answer_vocab is None:
+            self.answer_vocab = None
+        else:
+            if isinstance(answer_vocab, str):
+                answer_vocab = cached_path(answer_vocab, extract_archive=True)
+                answer_vocab = Vocabulary.from_files(answer_vocab)
+            self.answer_vocab = frozenset(
+                utils.preprocess_answer(a)
+                for a in answer_vocab.get_token_to_index_vocabulary("answers").keys()
+            )
+
+        if self.produce_featurized_images:
+            # normalize self.images some more
+            # At this point, self.images maps filenames to full paths, but we want to map image ids to full paths.
+            filename_re = re.compile(r".*(\d{12})\.((jpg)|(png))")
+
+            def id_from_filename(filename: str) -> Optional[int]:
+                match = filename_re.fullmatch(filename)
+                if match is None:
+                    return None
+                return int(match.group(1))
```

Comment:
What happened here? This never runs.

Reply:
My bad, that should've been deleted. It was left over from when I repurposed part of the vqa reader.